tools/scanTopology.cc

```diff
@@ -0,0 +1,122 @@
+#include "kickcat/Bus.h"
+#include "kickcat/Diagnostics.h"
+#include "kickcat/protocol.h"
+
+#ifdef __linux__
+    #include "kickcat/OS/Linux/Socket.h"
+#elif __PikeOS__
+    #include "kickcat/OS/PikeOS/Socket.h"
+#else
+    #error "Unknown platform"
+#endif
+
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <algorithm>
+
+using namespace kickcat;
+
+int main(int argc, char* argv[])
+{
+    if (argc != 2)
+    {
+        printf("usage: ./test NIC\n");
+        return 1;
+    }
+
+    auto socket = std::make_shared<Socket>();
+    Bus bus(socket);
+
+    auto print_current_state = [&]()
+    {
+        for (auto& slave : bus.slaves())
+        {
+            State state = bus.getCurrentState(slave);
+            printf("Slave %d state is %s\n", slave.address, toString(state));
+        }
+    };
+
+    uint8_t io_buffer[2048];
+    try
+    {
+        socket->open(argv[1], 2ms);
+        bus.init();
+
+        print_current_state();
+
+        bus.createMapping(io_buffer);
```

Comment:
Since in this example we only need to access to the DLstatus I don't think we need more than the bus init to be able to access the data.